Migrate gitlib::push_master to golang

puerco opened the pull request:
#### What type of PR is this?
/kind cleanup
/kind feature

#### What this PR does / why we need it:

This PR removes the `gitlib::push_master` and the call to it from `anago` and implements the same functions in go. The PR is split in three parts:

1. Four new functions are added to the git package:
    - `git.Fetch()` to fetch objects from a remote repository
    - `git.ShowLastCommit()` returns the last commit data from the log
    - `git.Rebase()` reabases to the specified branch
    - Finally, `git.Status()` returns a go-git status object with the current repo state
  Tests are included for each of these new functions

2. A new function has been added to the git pusher object: `PushMain()`. This function replaces `gitlib::push_master` by outputting some info, fetching, rebasing the main branch and finally pushing to the remote repo.
3. The third commit removes the bash code from anago and gitlib.sh

#### Which issue(s) this PR fixes:
Related to #1673

#### Special notes for your reviewer:

A test run of the code would be greatly appreciated !

#### Does this PR introduce a user-facing change?

```release-note
- The git package now has four new functions: `git.Status()`, `git.Fetch()`, `git.ShowLastCommit()` and `git.Rebase()`
- `krel anago ` now has a new flag `--push-main` that pushes changes made to the main branch
```

saschagrunert commented on cmd/krel/cmd/anago/push_git_objects.go:
```diff
@@ -104,6 +105,13 @@ func init() {
 		"number of times to retry git operations if a recoverable error occurs",
 	)
 
+	pushGitObjectsCmd.PersistentFlags().BoolVar(
+		&pushGitObjectsOpts.pushMain,
+		"push-main",
+		true,
+		"push changes in the main branch too",
+	)
```
I'm wondering why we need this if it defaults to `true` and we do not set it conditionally in `anago`

saschagrunert commented:
#### Manual testing
- :heavy_check_mark: Stage: https://console.cloud.google.com/cloud-build/builds/d5802279-8279-40ef-8640-11f47a171832?project=648026197307
- :heavy_check_mark: Release: https://console.cloud.google.com/cloud-build/builds/d29b80fb-51f2-45bc-9be2-a1df216e9edd?project=648026197307

Relevant seems only the release step:
```
Step #2: ================================================================================
Step #2: PUSH GIT OBJECTS  (11/14)
Step #2: ================================================================================
Step #2: 
Step #2: 
Step #2: anago::push_git_objects(): krel anago push-git-objects --release-branch=master --parent-branch= --repo=/workspace/anago-v1.20.0-beta.2/src/k8s.io/kubernetes --tags=v1.20.0-beta.2
Step #2: INFO Checkout master branch to push objects       
Step #2: INFO Pushing (mock) 1 tags                        
Step #2: INFO Pushing --dry-run tag for version v1.20.0-beta.2 
Step #2: INFO Won't push due to dry run repository         
Step #2: To https://github.com/kubernetes/kubernetes
Step #2:  * [new tag]                 v1.20.0-beta.2 -> v1.20.0-beta.2
Step #2: INFO Successfully pushed tag v1.20.0-beta.2       
Step #2: INFO Checkout master branch to push objects       
Step #2: INFO                                              
Step #2: INFO commit 99592159d04df33b034fc1a325baf41107481e6b
Step #2: Author: Anago GCB <nobody@k8s.io>
Step #2: Date:   Thu Nov 5 14:35:27 2020 +0000
Step #2: 
Step #2:     CHANGELOG: Update directory for v1.20.0-beta.2 release
Step #2: 
Step #2: diff --git a/CHANGELOG/CHANGELOG-1.20.md b/CHANGELOG/CHANGELOG-1.20.md
Step #2: index 9d2f166b02b..06e83ea5ea7 100644
Step #2: --- a/CHANGELOG/CHANGELOG-1.20.md
Step #2: +++ b/CHANGELOG/CHANGELOG-1.20.md
Step #2: @@ -1,69 +1,68 @@
Step #2:  <!-- BEGIN MUNGE: GENERATED_TOC -->
Step #2:  
Step #2: -- [v1.20.0-beta.1](#v1200-beta1)
Step #2: -  - [Downloads for v1.20.0-beta.1](#downloads-for-v1200-beta1)
Step #2: +- [v1.20.0-beta.2](#v1200-beta2)
Step #2: +  - [Downloads for v1.20.0-beta.2](#downloads-for-v1200-beta2)
Step #2:      - [Source Code](#source-code)
Step #2:      - [Client binaries](#client-binaries)
Step #2:      - [Server binaries](#server-binaries)
Step #2:      - [Node binaries](#node-binaries)
Step #2: -  - [Changelog since v1.20.0-beta.0](#changelog-since-v1200-beta0)
Step #2: +  - [Changelog since v1.20.0-beta.1](#changelog-since-v1200-beta1)
Step #2:    - [Changes by Kind](#changes-by-kind)
Step #2:      - [Deprecation](#deprecation)
Step #2: -    - [API Change](#api-change)
Step #2:      - [Feature](#feature)
Step #2: -    - [Documentation](#documentation)
Step #2:      - [Bug or Regression](#bug-or-regression)
Step #2:      - [Other (Cleanup or Flake)](#other-cleanup-or-flake)
Step #2:    - [Dependencies](#dependencies)
Step #2:      - [Added](#added)
Step #2:      - [Changed](#changed)
Step #2:      - [Removed](#removed)
Step #2: -- [v1.20.0-beta.0](#v1200-beta0)
Step #2: -  - [Downloads for v1.20.0-beta.0](#downloads-for-v1200-beta0)
Step #2: +- [v1.20.0-beta.1](#v1200-beta1)
Step #2: +  - [Downloads for v1.20.0-beta.1](#downloads-for-v1200-beta1)
Step #2:      - [Source Code](#source-code-1)
Step #2:      - [Client binaries](#client-binaries-1)
Step #2:      - [Server binaries](#server-binaries-1)
Step #2:      - [Node binaries](#node-binaries-1)
Step #2: -  - [Changelog since v1.20.0-alpha.3](#changelog-since-v1200-alpha3)
Step #2: -  - [Urgent Upgrade Notes](#urgent-upgrade-notes)
Step #2: -    - [(No, really, you MUST read this before you upgrade)](#no-really-you-must-read-this-before-you-upgrade)
Step #2: +  - [Changelog since v1.20.0-beta.0](#changelog-since-v1200-beta0)
Step #2:    - [Changes by Kind](#changes-by-kind-1)
Step #2:      - [Deprecation](#deprecation-1)
Step #2: -    - [API Change](#api-change-1)
Step #2: +    - [API Change](#api-change)
Step #2:      - [Feature](#feature-1)
Step #2: -    - [Documentation](#documentation-1)
Step #2: +    - [Documentation](#documentation)
Step #2:      - [Bug or Regression](#bug-or-regression-1)
Step #2:      - [Other (Cleanup or Flake)](#other-cleanup-or-flake-1)
Step #2:    - [Dependencies](#dependencies-1)
Step #2:      - [Added](#added-1)
Step #2:      - [Changed](#changed-1)
Step #2:      - [Removed](#removed-1)
Step #2: -- [v1.20.0-alpha.3](#v1200-alpha3)
Step #2: -  - [Downloads for v1.20.0-alpha.3](#downloads-for-v1200-alpha3)
Step #2: +- [v1.20.0-beta.0](#v1200-beta0)
Step #2: +  - [Downloads for v1.20.0-beta.0](#downloads-for-v1200-beta0)
Step #2:      - [Source Code](#source-code-2)
Step #2:      - [Client binaries](#client-binaries-2)
Step #2:      - [Server binaries](#server-binaries-2)
Step #2:      - [Node binaries](#node-binaries-2)
Step #2: -  - [Changelog since v1.20.0-alpha.2](#changelog-since-v1200-alpha2)
Step #2: +  - [Changelog since v1.20.0-alpha.3](#changelog-since-v1200-alpha3)
Step #2: +  - [Urgent Upgrade Notes](#urgent-upgrade-notes)
Step #2: +    - [(No, really, you MUST read this before you upgrade)](#no-really-you-must-read-this-before-you-upgrade)
Step #2:    - [Changes by Kind](#changes-by-kind-2)
Step #2: -    - [API Change](#api-change-2)
Step #2: +    - [Deprecation](#deprecation-2)
Step #2: +    - [API Change](#api-change-1)
Step #2:      - [Feature](#feature-2)
Step #2: +    - [Documentation](#documentation-1)
Step #2:      - [Bug or Regression](#bug-or-regression-2)
Step #2:      - [Other (Cleanup or Flake)](#other-cleanup-or-flake-2)
Step #2:    - [Dependencies](#dependencies-2)
Step #2:      - [Added](#added-2)
Step #2:      - [Changed](#changed-2)
Step #2:      - [Removed](#removed-2)
Step #2: -- [v1.20.0-alpha.2](#v1200-alpha2)
Step #2: -  - [Downloads for v1.20.0-alpha.2](#downloads-for-v1200-alpha2)
Step #2: +- [v1.20.0-alpha.3](#v1200-alpha3)
Step #2: +  - [Downloads for v1.20.0-alpha.3](#downloads-for-v1200-alpha3)
Step #2:      - [Source Code](#source-code-3)
Step #2:      - [Client binaries](#client-binaries-3)
Step #2:      - [Server binaries](#server-binaries-3)
Step #2:      - [Node binaries](#node-binaries-3)
Step #2: -  - [Changelog since v1.20.0-alpha.1](#changelog-since-v1200-alpha1)
Step #2: +  - [Changelog since v1.20.0-alpha.2](#changelog-since-v1200-alpha2)
Step #2:    - [Changes by Kind](#changes-by-kind-3)
Step #2: -    - [Deprecation](#deprecation-2)
Step #2: -    - [API Change](#api-change-3)
Step #2: +    - [API Change](#api-change-2)
Step #2:      - [Feature](#feature-3)
Step #2:      - [Bug or Regression](#bug-or-regression-3)
Step #2:      - [Other (Cleanup or Flake)](#other-cleanup-or-flake-3)
Step #2: @@ -71,30 +70,134 @@
Step #2:      - [Added](#added-3)
Step #2:      - [Changed](#changed-3)
Step #2:      - [Removed](#removed-3)
Step #2: -- [v1.20.0-alpha.1](#v1200-alpha1)
Step #2: -  - [Downloads for v1.20.0-alpha.1](#downloads-for-v1200-alpha1)
Step #2: +- [v1.20.0-alpha.2](#v1200-alpha2)
Step #2: +  - [Downloads for v1.20.0-alpha.2](#downloads-for-v1200-alpha2)
Step #2:      - [Source Code](#source-code-4)
Step #2:      - [Client binaries](#client-binaries-4)
Step #2:      - [Server binaries](#server-binaries-4)
Step #2:      - [Node binaries](#node-binaries-4)
Step #2: -  - [Changelog since v1.20.0-alpha.0](#changelog-since-v1200-alpha0)
Step #2: -  - [Urgent Upgrade Notes](#urgent-upgrade-notes-1)
Step #2: -    - [(No, really, you MUST read this before you upgrade)](#no-really-you-must-read-this-before-you-upgrade-1)
Step #2: +  - [Changelog since v1.20.0-alpha.1](#changelog-since-v1200-alpha1)
Step #2:    - [Changes by Kind](#changes-by-kind-4)
Step #2:      - [Deprecation](#deprecation-3)
Step #2: -    - [API Change](#api-change-4)
Step #2: +    - [API Change](#api-change-3)
Step #2:      - [Feature](#feature-4)
Step #2: -    - [Documentation](#documentation-2)
Step #2: -    - [Failing Test](#failing-test)
Step #2:      - [Bug or Regression](#bug-or-regression-4)
Step #2:      - [Other (Cleanup or Flake)](#other-cleanup-or-flake-4)
Step #2:    - [Dependencies](#dependencies-4)
Step #2:      - [Added](#added-4)
Step #2:      - [Changed](#changed-4)
Step #2:      - [Removed](#removed-4)
Step #2: +- [v1.20.0-alpha.1](#v1200-alpha1)
Step #2: +  - [Downloads for v1.20.0-alpha.1](#downloads-for-v1200-alpha1)
Step #2: +    - [Source Code](#source-code-5)
Step #2: +    - [Client binaries](#client-binaries-5)
Step #2: +    - [Server binaries](#server-binaries-5)
Step #2: +    - [Node binaries](#node-binaries-5)
Step #2: +  - [Changelog since v1.20.0-alpha.0](#changelog-since-v1200-alpha0)
Step #2: +  - [Urgent Upgrade Notes](#urgent-upgrade-notes-1)
Step #2: +    - [(No, really, you MUST read this before you upgrade)](#no-really-you-must-read-this-before-you-upgrade-1)
Step #2: +  - [Changes by Kind](#changes-by-kind-5)
Step #2: +    - [Deprecation](#deprecation-4)
Step #2: +    - [API Change](#api-change-4)
Step #2: +    - [Feature](#feature-5)
Step #2: +    - [Documentation](#documentation-2)
Step #2: +    - [Failing Test](#failing-test)
Step #2: +    - [Bug or Regression](#bug-or-regression-5)
Step #2: +    - [Other (Cleanup or Flake)](#other-cleanup-or-flake-5)
Step #2: +  - [Dependencies](#dependencies-5)
Step #2: +    - [Added](#added-5)
Step #2: +    - [Changed](#changed-5)
Step #2: +    - [Removed](#removed-5)
Step #2:  
Step #2:  <!-- END MUNGE: GENERATED_TOC -->
Step #2:  
Step #2: +# v1.20.0-beta.2
Step #2: +
Step #2: +
Step #2: +## Downloads for v1.20.0-beta.2
Step #2: +
Step #2: +### Source Code
Step #2: +
Step #2: +filename | sha512 hash
Step #2: +-------- | -----------
Step #2: +[kubernetes.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes.tar.gz) | 3e7cfd0dabe5c1bdb3067659133103a1bd5d9ade047eb9cfadfa50ac71b436af7df4daed1cd09c5860a10290fabcba4af495c553a7b9fba7d99fe7daa5e77d02
Step #2: +[kubernetes-src.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-src.tar.gz) | d97051c1873846e0e701bb790f44d1b8bdbfa954cfed96bd0f9cfd5e5646d6671a3b81f755d042c9d4d38d253686be4ecc9d8ecc08dd62191d090fae39b2aa78
Step #2: +
Step #2: +### Client binaries
Step #2: +
Step #2: +filename | sha512 hash
Step #2: +-------- | -----------
Step #2: +[kubernetes-client-darwin-amd64.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-client-darwin-amd64.tar.gz) | ebe0a0a4c4fa1d158b1006c2c31937c5b681ebee70fd0d181293bb858418b07dc447354917caa64ec763e72e4c3d48c5770f6dc9e5c8a438cc9f11907916d159
Step #2: +[kubernetes-client-linux-386.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-client-linux-386.tar.gz) | 9b21912d045019254313375c475f4a79888ac68b9c38b31d545ab2628631f9d01e225085e582eb6112320d2fb9a684f7d3198de18b5b12623fe5016d5ae088c5
Step #2: +[kubernetes-client-linux-amd64.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-client-linux-amd64.tar.gz) | 258166691e03c248eea5f16611eb5064099e0a183921c64ed55a40ac8b6eaf41870c70feb2538eeeefe9f042afb1c756caa4bc6a1abf9aaf0c0a84f847e1359e
Step #2: +[kubernetes-client-linux-arm.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-client-linux-arm.tar.gz) | a3412355026086cb2c8da8d29015ed64030c89df116a21cb027337f0168782bc7813d1bc40147efd079cee5027261c53e4301f2c9ca6af55dfb02c939b351446
Step #2: +[kubernetes-client-linux-arm64.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-client-linux-arm64.tar.gz) | eed275e256700bd098af7945d829fa67934368d12e62aabf66e14aae225862c1400c9cfa65be16a8c71bc422c46d6965330e93097a1c3beb3db8b37d6ae2832d
Step #2: +[kubernetes-client-linux-ppc64le.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-client-linux-ppc64le.tar.gz) | 799752d42860eff63c4b95e94ad910f90f2f8f85e8367aa5da8cff90d4e5a4b0931986202da43bc576fc8c842aaf2c049db01f3da6d5b25a3696bd58a87baf58
Step #2: +[kubernetes-client-linux-s390x.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-client-linux-s390x.tar.gz) | a2b7e1906b032cd8ff5acc6ef5c5c61b92048313822c12baa36566143532070d0d5a1d3c40e0869562a4d6e0658000c7d4bff02b49ed2ec2d948193221bdee36
Step #2: +[kubernetes-client-windows-386.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-client-windows-386.tar.gz) | e0ed32ceda55b3ff96df50a48a170f8818d13a9fa47f8fb07e73efb85a58725ef85aac2655590d1a8e57c610090195dc44644e76190bd9a74e0bbdfac9179b03
Step #2: +[kubernetes-client-windows-amd64.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-client-windows-amd64.tar.gz) | f768a9b766d3004efe35d11f274fea017771c130194ac3aef54a7a6be518b3bb824e48a7d6f950263f5cf8701d0fb78187d3a7cc38a2a04e96950b99eb1de766
Step #2: +
Step #2: +### Server binaries
Step #2: +
Step #2: +filename | sha512 hash
Step #2: +-------- | -----------
Step #2: +[kubernetes-server-linux-amd64.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-server-linux-amd64.tar.gz) | 197faf303b25b985df47613e37f506cd9ab3831ae3e43612122cb15e254c66587d8837102e16b6c3e2e5c6fa2733332296bc3282b64801d5d1c91cc2729f78c0
Step #2: +[kubernetes-server-linux-arm.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-server-linux-arm.tar.gz) | ed94bf9c46698a87628de38fb491bb49e3837b1ca4003181b7eed61cb67e3de9d1e82c3f342468e90eca58d038e136288bfb9f538ba7b9647623c753b3c4b601
Step #2: +[kubernetes-server-linux-arm64.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-server-linux-arm64.tar.gz) | e15e4af78e6c8da181dad77fde34bfe0a38f608c7ba0654c56539c47093a991d96471a5f4ea776421921f18f74e760f5dc9b865987e3c673874b964dadf8cccb
Step #2: +[kubernetes-server-linux-ppc64le.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-server-linux-ppc64le.tar.gz) | 232f36cb22abb79f122db8ec1b59e008b98f343430d3c6bab37486dfee607a80976abba94825c07e3222ec18cd955d1900288728c187cbf4f01413f064d3d392
Step #2: +[kubernetes-server-linux-s390x.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-server-linux-s390x.tar.gz) | 517736e58a0e32b14e4fdb3267755b86853d391dbfe3433cd214ca5acee2f8e3bc28c8a53205c26ad940fc669f99479919b4ff820676610ad57f4e7d38c981ed
Step #2: +
Step #2: +### Node binaries
Step #2: +
Step #2: +filename | sha512 hash
Step #2: +-------- | -----------
Step #2: +[kubernetes-node-linux-amd64.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-node-linux-amd64.tar.gz) | 372128552436fd00969e64622c3bdd546915402acead92faa0db2dfd89d3af00cba332d995e7c7443c099e7351870b950b0d531fecf871f3e566a511aba765e0
Step #2: +[kubernetes-node-linux-arm.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-node-linux-arm.tar.gz) | b64aa3f4fe926e20a24c5071ea3d7496cd0c8cfe36028c54fd6218c1271cda6d10119b478a0217ff39d5c11d133fe26b184d4e4ff43dd03f978531a82614abc4
Step #2: +[kubernetes-node-linux-arm64.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-node-linux-arm64.tar.gz) | 8b3116a30bc0ec2d280be7e01f825f58c30c240dc7f287d1b967eb9ba5de3a9035bddcde64abef39d7a70a999de489be17a68931b3068a2f40a595d4ca0a3302
Step #2: +[kubernetes-node-linux-ppc64le.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-node-linux-ppc64le.tar.gz) | 1026ca779bc88571cfe189aec329c87018b7bb469a2cc4a25ffb34af408a9b430b87059a415501d070d54b8efbea8191b88ea070d62556bf902f1cc8f456195e
Step #2: +[kubernetes-node-linux-s390x.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-node-linux-s390x.tar.gz) | d6638f315386dd6048b38034d8b127cc7492bf0be3cd8d0a8a70bce89cbda969bd7dc0eb9464fb75aa525dfcdc1d3cb1d86a24f80348934f1d341cdbfeb34276
Step #2: +[kubernetes-node-windows-amd64.tar.gz](https://storage.googleapis.com/kubernetes-release-gcb/v1.20.0-beta.2/kubernetes-node-windows-amd64.tar.gz) | aee54ecca48ac5030dcb4217e6eccf6171d3be0258477dcb6840103dd540967af3ee4be9b1ace02be7058c842b16ee033560677294beaac4ea5f379b4b778730
Step #2: +
Step #2: +## Changelog since v1.20.0-beta.1
Step #2: +
Step #2: +## Changes by Kind
Step #2: +
Step #2: +### Deprecation
Step #2: +
Step #2: +- Kubectl: deprecate --delete-local-data ([#95076](https://github.com/kubernetes/kubernetes/pull/95076), [@dougsland](https://github.com/dougsland)) [SIG CLI, Cloud Provider and Scalability]
Step #2: +
Step #2: +### Feature
Step #2: +
Step #2: +- Introduce api-extensions category which will return: mutating admission configs, validating admission configs, CRDs and APIServices when used in kubectl get, for example. ([#95603](https://github.com/kubernetes/kubernetes/pull/95603), [@soltysh](https://github.com/soltysh)) [SIG API Machinery]
Step #2: +- Kube-apiserver: The timeout used when making health check calls to etcd can now be configured with `--etcd-healthcheck-timeout`. The default timeout is 2 seconds, matching the previous behavior. ([#93244](https://github.com/kubernetes/kubernetes/pull/93244), [@Sh4d1](https://github.com/Sh4d1)) [SIG API Machinery]
Step #2: +- The beta `RootCAConfigMap` feature gate is enabled by default and causes kube-controller-manager to publish a "kube-root-ca.crt" ConfigMap to every namespace. This ConfigMap contains a CA bundle used for verifying connections to the kube-apiserver. ([#96197](https://github.com/kubernetes/kubernetes/pull/96197), [@zshihang](https://github.com/zshihang)) [SIG API Machinery, Apps, Auth and Testing]
Step #2: +- `kubectl alpha debug` has graduated to beta and is now `kubectl debug`. ([#96138](https://github.com/kubernetes/kubernetes/pull/96138), [@verb](https://github.com/verb)) [SIG CLI and Testing]
Step #2: +- `kubectl debug` gains support for changing container images when copying a pod for debugging, similar to how `kubectl set image` works. See `kubectl help debug` for more information. ([#96058](https://github.com/kubernetes/kubernetes/pull/96058), [@verb](https://github.com/verb)) [SIG CLI]
Step #2: +
Step #2: +### Bug or Regression
Step #2: +
Step #2: +- Change plugin name in fsgroupapplymetrics of csi and flexvolume to distinguish different driver ([#95892](https://github.com/kubernetes/kubernetes/pull/95892), [@JornShen](https://github.com/JornShen)) [SIG Instrumentation, Storage and Testing]
Step #2: +- Fix bug in JSON path parser where an error occurs when a range is empty ([#95933](https://github.com/kubernetes/kubernetes/pull/95933), [@brianpursley](https://github.com/brianpursley)) [SIG API Machinery]
Step #2: +- Skip [k8s.io/kubernetes@v1.19.0/test/e2e/storage/testsuites/base.go:162]: Driver azure-disk doesn't support snapshot type DynamicSnapshot -- skipping
Step #2: +  skip [k8s.io/kubernetes@v1.19.0/test/e2e/storage/testsuites/base.go:185]: Driver azure-disk doesn't support ntfs -- skipping ([#96144](https://github.com/kubernetes/kubernetes/pull/96144), [@qinpingli](https://github.com/qinpingli)) [SIG Storage and Testing]
Step #2: +
Step #2: +### Other (Cleanup or Flake)
Step #2: +
Step #2: +- NONE ([#96179](https://github.com/kubernetes/kubernetes/pull/96179), [@bbyrne5](https://github.com/bbyrne5)) [SIG Network]
Step #2: +
Step #2: +## Dependencies
Step #2: +
Step #2: +### Added
Step #2: +_Nothing has changed._
Step #2: +
Step #2: +### Changed
Step #2: +- k8s.io/klog/v2: v2.2.0 → v2.4.0
Step #2: +
Step #2: +### Removed
Step #2: +_Nothing has changed._
Step #2: +
Step #2: +
Step #2: +
Step #2:  # v1.20.0-beta.1 
Step #2: INFO Rebase master branch                         
Step #2: INFO Rebasing repository to origin/master         
Step #2: INFO Pushing --dry-run master branch              
Step #2: INFO Won't push due to dry run repository         
Step #2: To https://github.com/kubernetes/kubernetes
Step #2:    bd95fb101dd..3b1d1ffe177  master -> master
Step #2: INFO git objects push complete                    
Step #2: [2020-Nov-06 08:27:54 UTC] push_git_objects in 20s
```

puerco commented:
Bad habits I guess, even when I know that this will run headless in GCB, I don't know why I feel a need to have a way to control all inputs to the steps controllable from the CLI.

I sometimes think that someone might need to go in and re-call a certain part. Of course this is not needed at all if we succeed in building stage and release completely reentrant and idempotent. 

I've removed the option and added a specific `Info()` call  here, when repo.Status() returns clean:
```
Step #2: INFO Checkout master branch to push objects       
Step #2: INFO                                              
Step #2: INFO commit 99592159d04df33b034fc1a325baf41107481e6b
```

saschagrunert reviewed (approved):
/lgtm

saschagrunert commented:
/lgtm
